crates/wast/src/resolve/names.rs

```diff
@@ -1675,18 +1688,26 @@ impl<'a, T> Default for Namespace<'a, T> {
     }
 }
 
+#[derive(Debug, Clone)]
+struct ExprBlock<'a> {
+    // The label of the block
+    label: Option<Id<'a>>,
+    // Whether this block pushed a new scope for resolving locals
+    pushed_scope: bool,
+}
+
 struct ExprResolver<'a, 'b> {
     module: &'b Module<'a>,
-    locals: Namespace<'a, ()>,
-    labels: Vec<Option<Id<'a>>>,
+    scopes: Vec<Namespace<'a, ()>>,
```

Comment:
Mind adding a comment to this saying that the scopes here are for locals and that this is dynamically managed as we exit/enter `let` blocks?